packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -234,9 +224,7 @@ export class TokenRatesController extends StaticIntervalPollingControllerV1<
     onPreferencesStateChange(async ({ selectedAddress }) => {
       if (this.config.selectedAddress !== selectedAddress) {
         this.configure({ selectedAddress });
-        if (this.#pollState === PollState.Active) {
```

Comment:
Not sure about the purpose of these `PollState.Active` guards.  It seems like we already have `disabled` meant to guard the setting being disabled.  So I'm not sure why else we'd guard this.  I was hoping to not re-create this enum/state since I don't think it maps cleanly to multichain polling.

Reply:
If we're removing the legacy polling, then `updateExchangeRates` itself is no longer relevant. That method was just for updating the exchange rates for the selected network, which is also what the legacy poller is tracking.

With the legacy polling eliminated, we don't want to treat the selected network as a special case anymore. We shouldn't be asking for that data unless there is an active poll.

That said, it does seem appropriate to restart any active polls at this point, so they don't have stale data until the next interval.

Reply:
Hmmm `updateExchangeRates` seemed still relevant because we have to trigger when the account changes.  Since prices are only queried for tokens owned by the current account.

I'm not sure how to better handle account changes with the new polling mechanism.  If we don't keep these existing subscriptions within the controller that call `updateExchangeRates`, I suppose we could try moving account switch handling to the client.  It could maybe pass the current account to the `startPollingByNetworkClientId` options.

Reply:
Hmmm we need a pattern to handle account switches with the new polling mechanism.  For controllers like this + detection that are scoped to the current account.  Two ideas:

1. Like before.  The controller keeps maintaining state and subscriptions for the currently selected address.  When it changes, it updates its address state (which affects the next poll), but also calls the one off `updateExchangeRates` (or equivalent) so we don't wait until next poll.

2. The client could become responsible for handling account switches, instead of the controller.  On account switch, it would do `stopPollingByPollingToken` followed by `startPollingByNetworkClientId` that includes the current address in the `options` parameter.

Also it could be good to debounce/cache these chain + account switch triggers, since they currently run on every change.  It could be smarter and know "we just fetched for that chain + account 30 seconds ago".